### ISSUE TITLE
 httpNodeAuth  with  the dashboard path.

### ISSUE THREAD

**Issue by vcailly** [node-red/node-red-dashboard#384]
I  try to set  httpNodeAuth to password protect the node-defined HTTP endpoints 

` httpNodeAuth: {user:"user",pass:"$2a$08$Yoi2hNXHuclRT2oJN5tpC.CvxRCscs5..03UI.*************"}`

Dashboard is storing data  in localSTorage ( localStorage stores information as long as the user does not delete them.),   

Once  data has been stored in the browser's localStorage, authentication is no more required 

Authentication is required only once (unless you clear  manually the localStorage  of the browser).   In my humble opinion, It is not really secured.

Is there a way to have the dashboard  using sessionStorage (sessionStorage stores information as long as the session goes. Usually until the user closes the tab/browser),  instead of localStorage .

**Comment by dceejay** [comment]
Hi,
not quite sure what is happening here. Are you saying that because it is storing some data in localStorage then the session never ends ? Or that once you log out, that the data is still there  and this may be insecure ?

The only thing stored is the state of the groups (open or closed) so not particularly "leaky" - but I guess it could all be used to mean something. If this was stored in sessionStorage then next time you connect the groups would be back to default state - which is what this was trying to avoid.

The reality of course is that simple httpAuth is probably not the best way to handle this at all - and that with another auth scheme then the ui state data could potentially be held at the backend instead.

**Comment by vcailly** [comment]
It looks like the session never ending.   The  authentication windows popups only after clearing all  "site data".   I observe this behavior on both Firefox and chrome. 

I am using:
node  v6.13.1
node-red   v0.18.4
node-red-dashboard v2.9.0

**Comment by Paul-Reed** [comment]
If you clear the browser application cache it will force a re-auth.
In chrome, use the url - [chrome://appcache-internals/](chrome://appcache-internals/) to see and delete the respective cache.

**Comment by dceejay** [comment]
Hmm actually reverting this fix as localStorage does the right thing from the UX point of view. A better fix for this would be to allow the dashboard user to log out properly and thence delete localStorage at that point....  so re-opening for some time to think about it.

**Comment by sslupsky** [comment]
I am not sure if I am having the same problem so maybe this needs to be a new issue.  I enabled adminAuth and httpNodeAuth.  AdminAuth is working fine (tested with Safari).  httpNodeAuth is not.  Using Safari, no authentication is required to login.  Using FireFox, I get message requesting credentials for the "AutoAuth Extension" ... I do not see the actual login request.  Using Chrome, I see the "Sign in" dialog.  I have attached a couple screenshots of Firefox and Chrome.

![screen shot 2018-05-16 at 9 03 26 am](https://user-images.githubusercontent.com/16888734/40125589-8786e892-58e8-11e8-900f-bf9f6d3e72a7.png)
<img width="633" alt="screen shot 2018-05-16 at 9 03 42 am" src="https://user-images.githubusercontent.com/16888734/40125601-8a5a393e-58e8-11e8-8379-d894249e9555.png">

**Comment by dceejay** [comment]
Yes,
this seems to be a "feature" of basicAuth... most browsers seem to be too clever for themselves and keep hold of the credentials for the site - so when you go back they log you straight in.
There are some hacky ways to try to log out - and they do seem to work. I have tried adding a logout button that tries to login with a bad set of credentials and so then fails...  and it does seem to work but ONLY if you use your browser in incognito mode. Once again the browser seems to keep the older good credentials hanging around and re-uses them when it can... 

If anyone has any good ideas how to force basicAuth to logout then please shout.

**Comment by eppak** [comment]
Hello,
i have a similar problem, just put adminAuth in settings and restarting; the editor now ask for password but the /ui don't. Tried with private mode and a browser never accessed to this site. Any ideas?

Thanks

**Comment by dceejay** [comment]
the adminAuth and httpAuth are separate.
the Dashboard is not part of the Editor so is not protected by adminAuth

**Comment by bneumann** [comment]
Just stumbled upon that. The credentials for httpAuth are definitly cached. Any other way on securing the dashboard?
